pkg/grpc/exec/exec.go

```diff
@@ -288,6 +288,15 @@ func GetProcessExit(event *MsgExitEventUnix) *tetragon.ProcessExit {
 	code := event.Info.Code >> 8
 	signal := readerexec.Signal(event.Info.Code & 0xFF)
 
+	// Ensure that we get PID == TID
+	if event.ProcessKey.Pid != event.Info.Tid {
+		logger.GetLogger().WithFields(logrus.Fields{
+			"event.name":        "Exit",
+			"event.process.pid": event.ProcessKey.Pid,
+			"event.process.tid": event.Info.Tid,
+		}).Debug("ExitEvent: process PID and TID mismatch")
+	}
```

Comment:
the comment says you want to ensure PID == TID, but there's just warning,
we don't set TID = PID as we do for exec event

Reply:
Indeed @olsajiri , I added a comment why in the updates patches. But basically we re-use the TID of exec/clone event anyway. so even if we set it , it will be overwritten after.

This check is for debugging and I commented in the patch that we are still not finished with the complete per thread transition, as we still don't handle capabilities and namespaces, and remember we have only one entry per process threads that is the leader either in bpf execve_val map or in the user space process cache.

Later I will remove those extra TIDs that I'm sending from BPF in clone/execve and exit, and just send PID and re-use it in userspace as TID.